C/c4Document.cc

```diff
@@ -596,7 +638,17 @@ C4Document* c4doc_update(C4Document *doc,
 
 bool c4doc_removeRevisionBody(C4Document* doc) noexcept {
     auto idoc = asInternal(doc);
-    return idoc->mustBeInTransaction(NULL) && idoc->removeSelectedRevBody();
+    if(!idoc->mustBeInTransaction(nullptr)) {
+        WarnError("c4doc_removeRevisionBody: Called outside transaction");
+        return false;
+    }
+
+    if(!idoc->removeSelectedRevBody()) {
+        Warn("c4doc_removeRevisionBody: failed to remove");
```

Comment:
Not clear why this is worth a warning.

Reply:
I am heavily erring on the side of verbosity here.  I got tired of finding places where it would have been useful to know why a function returned a non-expected value and so I've been going through and filling them in.